File: 2444-count-subarrays-with-fixed-bounds/2444-count-subarrays-with-fixed-bounds.java
class Solution {
    public long countSubarrays(int[] nums, int minK, int maxK) {
        Set<Integer> set = new HashSet<>();
        int pivot = minK == maxK ? 1 : 2;
        Map<Integer, Integer> convert = new HashMap<>();
        Map<Integer, Integer> index = new HashMap<>();
        
        convert.put(minK, maxK);
        convert.put(maxK, minK);
        int outRange = -1;
        long answer = 0;
        int max = 0;
        
        for (int i = 0; i < nums.length; i++) {
            if (minK == nums[i] || maxK == nums[i]) {
                set.add(nums[i]);
                index.put(nums[i], i);
            }
            
            if (minK <= nums[i] && nums[i] <= maxK) {
                if (set.size() == pivot) {
                    if (minK == nums[i] || maxK == nums[i]) {
                        int reverse = convert.get(nums[i]);
                        max = Math.max(max, index.get(reverse) - outRange);
                        answer += index.get(reverse) - outRange;
                    } else {
                        answer += max;
                    }
                }
            } else {
                set.clear();
                index.clear();
                outRange = i;
                max = 0;
            }
        }
        
        return answer;
    }
}

File: 2444-count-subarrays-with-fixed-bounds/2444-count-subarrays-with-fixed-bounds.cpp
class Solution {
public:
  long long countSubarrays(vector<int>& nums, int minK, int maxK) {
    map<int,int>m;
    int n=nums.size();
    int lower=0;
    int l=0;
    long long answer=0;
    for(int i=0;i<n;i++){
      if(nums[i]<minK||maxK<nums[i]){
        m=map<int,int>();
        lower=i+1;
        l=i+1;
        continue;
      }
      m[nums[i]]++;
      while(l!=i&&(nums[l]!=minK&&nums[l]!=maxK)||1<m[nums[l]]){
        m[nums[l++]]--;
      }
      if(1<=m[minK]&&1<=m[maxK]){
        answer+=(l-lower+1);
      }
    }
    return answer;
  }
};